src/Parser/Rogue/Subtlety/Modules/CastTracker/FilteredDamageTracker.js

```diff
@@ -0,0 +1,25 @@
+import DamageTracker from 'Parser/Core/Modules/AbilityTracker';
+
+class FilteredDamageTracker extends DamageTracker {
+  
+  on_byPlayer_damage(event) {
+    if(!this.shouldProcessEvent(event)) return;
+    super.on_byPlayer_damage(event);
+  }
+
+  on_byPlayer_heal(event) {
+    if(!this.shouldProcessEvent(event)) return;
+    super.on_byPlayer_heal(event);
+  }
+
+  on_byPlayer_cast(event) {
+    if(!this.shouldProcessEvent(event)) return;
+    super.on_byPlayer_cast(event);
+  }
+    
+  shouldProcessEvent(event) {
+    return false;
+  }
```

Comment:
Instead of having this redirect unnecessarily, why don't you just move the onByPlayer_damage into the two classes that extend this one, and just have those extend the DamageTracker? You never call this one anyways - seems overly redundant to me.

Reply:
because it  would lead to duplicating the logic for filtering 3 types of events to both existing trackers. I don't understand what you mean by "You never call this one anyways" - its overridden in both FilteredDamageTracker and DanceDamageTracker

Reply:
Do you care about the heal events of those spells? Do they even have heal events, and of what use are they?

Reply:
I don't care about heal events, but the idea here is to filter out all events. Would make no sense to have a tracker that calculates everything during a certain condition, but ignores that condition for healing. I can disable healing aggregation by creating 3  "shouldProcess" functions and not overriding the healing one. Even if I drop healing its still casts and damage filtering that must be duplicated to every tracker. And I will need more - I will need more of these trackers down the line. My other argument is that I find SymbolsDamageTracker to be more concise when it only has 1  method overridden - shouldProcessEvent, instead of having duplicated logic between all trackers.